src/js/stores/DCOSStore.js

```diff
@@ -120,7 +123,13 @@ class DCOSStore extends EventEmitter {
     this.data.marathon.deploymentsList = deploymentsList
       .mapItems(function (deployment) {
         let ids = deployment.getAffectedServiceIds();
-        let services = ids.map(serviceTree.findItemById.bind(serviceTree));
+        let services = ids.reduce(function (memo, id) {
+          let service = serviceTree.findItemById(id);
```

Comment:
Hey @orlandohohmeier, sorry I didn't see this until just now. I agree that's a much nicer approach, let's make that change separately.